### PR TITLE
Small improvements mainly focusing on docs

### PR DESCRIPTION
**Pull request by abravalheri**
This PR brings some changes from #353, specially in terms of documentation.

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -6,7 +6,11 @@ Frequently Asked Questions
 
 In case you have a general question that is not answered here, consider submitting a `new issue`_.
```
Comment (FlorianWilhelm) [inline docs/faq.rst:7]: Should we rather refer users to the new *Discussions* feature of Github instead of Issues?

File: docs/faq.rst [hunk 2]
```diff
@@ -193,28 +184,80 @@ In case you have a general question that is not answered here, consider submitti
...
+    If you prefer to have legacy builds and get the old behavior, you can remove the ``pyproject.toml`` file and run
     ``python setup.py bdist_wheel``, but we advise to install the build requirements (as the ones specified in the
     ``requires`` field of ``pyproject.toml``) in an `isolated environment`_ and use it to run the ``setup.py`` commands
     (`tox`_ can be really useful for that). Alternatively you can use the ``setup_requires`` field in `setup.cfg`_,
-    however, this method is discouraged and might be invalid in the future. To skip the generation of the
-    ``pyproject.toml`` file you can run ``putup`` with the ``--no-pyproject`` flag.
+    however, this method is discouraged and might be invalid in the future.
+
+    .. warning::
+       For the time being you can use the **transitional** ``--no-pyproject``
```
Comment (FlorianWilhelm) [inline docs/faq.rst:208]: I was just thinking. Should a kind of [not recommended] statement also be added to the cli help of `--no-pyproject`?

**Review by FlorianWilhelm — APPROVED** [review]
lgtm

File: docs/faq.rst (hunk 2 above)
Comment (abravalheri) [reply docs/faq.rst:208]: That is a good call! I will merge the PR and then add it to master.

File: docs/faq.rst (hunk 1 above)
Comment (abravalheri) [reply docs/faq.rst:7]: I will also add it directly when merging to master, since it does not interfere with the CI.

**Pull request merged**